xskillscore/core/comparative.py

```diff
@@ -280,39 +295,79 @@ def mae_test(
     """
     if isinstance(dim, str):
         dim = [dim]
+    elif dim is None:
+        dim = []
```

Comment:
usually we try to follow the convention from `xarray`. if dim=None, the function is applied over all dimensions possible.

we didnt have this before, but it looks like we should have...

I see we had `dim=[]` in the definition. but I think having lists in the definition is also not good style.

Reply:
Yes, initializing with empty list is not good style, and some of the linters even complain if you do. If you prefer I can revert it to the way it was.

Reply:
ah. the problem is that if we use dim=None and pass None into the metrics, we reduce all dimensions. I think they way you did it here is good. see my comment for the docs